README.md

```diff
@@ -7,9 +7,45 @@
 The crawler identifies source repositories from registered [PURLs (Package URLs)][purl] and copies VEX documents into VEX Hub.
 This process ensures that VEX Hub maintains an up-to-date collection of VEX documents for various software packages.
 
+The following diagram illustrates the high-level process flow of the VEX Hub Crawler, using npm as an example:
+
+```mermaid
+flowchart TD
+    Dev[Developer] -->|1. Register PURLs| PL
+    PL[PURL List] -->|2. Provide PURLs for crawling| Crawler
+    Crawler --> Registry[npm Registry]
+    Registry -->|3. Identify source repository URL| Crawler
+    Crawler --> Src
+    Src -->|4. Retrieve VEX documents| Crawler
+    Crawler -->|5. Validate and copy VEX documents| Hub[VEX Hub]
+
+    subgraph Src [Source Repository]
+        Dir[.vex/ directory] -->|Store| Doc[VEX documents]
+    end
```

Comment:
not sure a subgraph helps here, maybe simplify with just a text that demonstrates the .vex dir path?
anyway, shouldn't it be `/.vex` instead of `.vex/`